src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/SourceTextExtensions.cs

```diff
@@ -220,22 +230,19 @@ public static bool TryGetAbsoluteIndex(this SourceText sourceText, int line, int
         }
         else
         {
-            absoluteIndex = sourceText.Lines[line].Start + character;
-        }
-
-        return true;
-    }
-
-    public static bool TryGetAbsoluteIndex(this SourceText sourceText, int line, int character, ILogger? logger, out int absoluteIndex)
-    {
-        if (!TryGetAbsoluteIndex(sourceText, line, character, out absoluteIndex))
-        {
+            var sourceLine = sourceText.Lines[line];
+            var lineLengthIncludingLineBreak = sourceLine.SpanIncludingLineBreak.Length;
+            if (character > lineLengthIncludingLineBreak)
+            {
 #pragma warning disable CA2254 // Template should be a static expression.
-            // This is actually static, the compiler just doesn't know it.
-            logger?.LogError(SR.FormatPositionIndex_Outside_Range(line, nameof(sourceText), sourceText.Lines.Count));
+                // This is actually static, the compiler just doesn't know it.
+                logger?.LogError(SR.FormatPositionCharacter_Outside_Range(character, nameof(sourceText), lineLengthIncludingLineBreak));
```

Comment:
I'm tempted to also Debug.Fail here when logger is not null so it would be more visible to devs if they hit it while developing/testing/app-building. Thoughts?